### PR TITLE
fix(forms): hide input control text under valid icon

### LINKED ISSUE #1305: Text overflow not hidden with input text validation icons
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Hi,
I encoutered the following issue.
To avoid implementation doubts, this issue is reproducible on Bootstrap 5.1 and Boosted 5.1 respective showcase sites.
When an input has more characters than input size, overflow goes on top of validation icons on boosted.
It doesn't behave the same with Bootstrap, overflow is hidden correctly

Capture d’écran 2022-06-10 à 13 14 37
Capture d’écran 2022-06-10 à 13 14 18

### Reduced test cases

Please test directly on Vendor platforms :
https://getbootstrap.com/docs/5.1/forms/validation/
https://boosted.orange.com/docs/5.1/forms/validation/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Boosted are you using?

5.1

### PR DESCRIPTION
**Pull request by julien-deramond**
Fixes #1305 

This PR is a proposal to fix #1305 by:
* adding the same `padding-right` rule than in Bootstrap
* avoiding to display the valid icon for Quantity Selector (not enough space)

Dear reviewer please check all possible cases in forms when inputs are valid.

### Live previews
* [Forms > Validation > Custom styles](https://deploy-preview-1306--boosted.netlify.app/docs/5.1/forms/validation/#custom-styles)
* [Forms > Validation > Supported elements](https://deploy-preview-1306--boosted.netlify.app/docs/5.1/forms/validation/#supported-elements)

### CODE REVIEW

File: scss/mixins/_forms.scss [hunk 1]
```diff
@@ -25,8 +25,9 @@
     }
 
     @if $enable-validation-icons {
-      .form-control {
+      :not(.quantity-selector) > .form-control {
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply scss/mixins/_forms.scss:28]: Before this PR, in https://boosted.netlify.app/docs/5.1/forms/validation/#supported-elements:
Capture d’écran 2022-06-13 à 07 57 26
If I don't exclude `.quantity-selector` the validation icon would be on top of the input value hiding it. I preferred to remove it entirely.

**Review by louismaximepiton — CHANGES_REQUESTED** [review]
The actual implementation seems to be great, but here are few questions:

Color picker could be removed too ? Or maybe do it as Bs ?
image
image
image
image


Few points that could be handled in other PR('s):

* We should remove `.is-invalid` l.249 in `validation.md`
* We should be careful for `input[type="switch"][required]:not(:checked)` invalid state (due to a filter):
image

**Review by MewenLeHo — CHANGES_REQUESTED** [review]
I think we still need the changes suggested by @louismaximepiton in his comment.
Or at least we should create a distinct issue for the problem raised by LM.

**Comment by julien-deramond** [comment]
@MewenLeHo or @louismaximepiton Please check this new version including https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1306/commits/87892ff3da92ffb5231512fb4632c2870db7e11a which should fix `.form-control-color` and `.form-check-input` use cases 🤞

**Comment by MewenLeHo** [comment]
> @MewenLeHo or @louismaximepiton Please check this new version including [87892ff](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/commit/87892ff3da92ffb5231512fb4632c2870db7e11a) which should fix `.form-control-color` and `.form-check-input` use cases 🤞

LGTM 👌

**Pull request merged**